apps/prairielearn/src/lib/externalGradingSocket.ts

```diff
@@ -101,52 +108,46 @@ export function connection(socket) {
   });
 }
 
-export function getVariantSubmissionsStatus(variant_id, callback) {
-  const params = {
-    variant_id,
-  };
-  sqldb.query(sql.select_submissions_for_variant, params, (err, result) => {
-    if (ERR(err, callback)) return;
-    callback(null, result.rows);
-  });
+export async function getVariantSubmissionsStatus(variant_id: string) {
+  return await sqldb.queryRows(
+    sql.select_submissions_for_variant,
+    { variant_id },
+    SubmissionForVariantSchema,
+  );
 }
 
-export function gradingJobStatusUpdated(grading_job_id) {
-  const params = { grading_job_id };
-  sqldb.queryOneRow(sql.select_submission_for_grading_job, params, (err, result) => {
-    if (
-      ERR(err, (err) => {
-        logger.error('Error selecting submission for grading job', err);
-        Sentry.captureException(err);
-      })
-    ) {
-      return;
-    }
+export async function gradingJobStatusUpdated(grading_job_id: string) {
+  try {
+    const submission = await sqldb.queryRow(
+      sql.select_submission_for_grading_job,
+      { grading_job_id },
+      SubmissionForGradingJobSchema,
+    );
+
     const eventData = {
-      variant_id: result.rows[0].variant_id,
-      submissions: result.rows,
+      variant_id: submission.variant_id,
+      submissions: [submission],
```

Comment:
I kept the signature of the object, but this is the only location where this event is emitted, so this could be changed to a single submission and changed in the code that handle this event (the question client code and a test) to accept a single submission.

Reply:
Good point. I'll leave it for another PR.